mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/ElasticSearchIndexer.java

```diff
@@ -53,6 +55,29 @@ public String getId() {
         public String getUpdatedDocumentPart() {
             return updatedDocumentPart;
         }
+
+        @Override
+        public final boolean equals(Object o) {
```

Comment:
Optimisation suggestion (but I think this is small), start with:
`if (this == o) {
   return true; 
 }`